main.rb

```diff
@@ -0,0 +1,220 @@
+# rubocop: disable Metrics/CyclomaticComplexity, Metrics/ModuleLength, Metrics/MethodLength, Metrics/PerceivedComplexity
+
+module Enumerable
+  def my_each
+    return enum_for(:my_each) unless block_given?
+
+    0.upto(to_a.size - 1) do |k|
+      yield to_a[k]
+    end
+    to_a
+  end
+
+  def my_each_with_index
+    return enum_for(:my_each) unless block_given?
+
+    0.upto(to_a.size - 1) do |k|
+      yield to_a[k], k
+    end
+    to_a
+  end
+
+  def my_select
+    return enum_for(:my_each) unless block_given?
+
+    result = []
+    0.upto(to_a.size - 1) do |k|
+      ret = yield to_a[k]
+      result << to_a[k] if ret == true
+    end
+    result
+  end
+
+  def my_all?(*args)
+    if block_given?
+      to_a.my_each do |item|
+        return false unless yield item
+      end
+    elsif args&.length&.positive? and to_a.length.positive?
+      if args[0].class == Regexp
+        to_a.my_each do |item|
+          return false unless item.to_s =~ args[0]
+        end
+      elsif args[0].class == Class
+        to_a.my_each do |item|
+          return false unless item.is_a?args[0]
+        end
+      else
+        to_a.my_each do |item|
+          return false unless item == args[0]
+        end
+      end
+    else
+      to_a.my_each do |item|
+        return false unless item
+      end
+    end
+    true
+  end
+
+  def my_any?(*args)
+    if block_given?
+      to_a.my_each do |item|
+        return true if yield item
+      end
+    elsif args&.length&.positive? and to_a.length.positive?
+      if args[0].class == Regexp
+        to_a.my_each do |item|
+          return true if item.to_s =~ args[0]
+        end
+      elsif args[0].class == Class
+        to_a.my_each do |item|
+          return true if item.is_a?args[0]
+        end
+      else
+        to_a.my_each do |item|
+          return true if item == args[0]
+        end
+      end
+    else
+      to_a.my_each do |item|
+        return true if item
+      end
+    end
+    false
+  end
+
+  def my_none?(*args)
```

Comment:
- ```my_none?``` and ```none?``` must return same results in the examples below
```%w{ant bear cat}.my_none?{ |word| word.length >=4}  != %w{ant bear cat}.none?{ |word| word.length >= 4 }```
``` [1, 3.14, 42].my_none?(Float) != [1, 3.14, 42].none?(Float)```
``` [ ].my_none? `!=` [ ].none? ```
``` [nil, false, true].my_none? `!=`  [nil, false, true].none? ```

Reply:
Thanks for your review and time @Forison. Hope you are having an awesome time. I took my time and indeed found some unattended edge cases on the methods you mentioned above. I went further and fixed the bugs and tested it. I can firmly say that the methods are working as expected.
 Thank you.